### PR TITLE
Added polynomial lr scheduler for training

### PR DESCRIPTION
**Pull request by ibadami**
Hey I just saw your self note for polynomial learning rate scheduler like in PSPnet. I happen to have it in my own training code. In case you want to save 5 minutes :)

### CODE REVIEW

**Comment by meetshah1995** [comment]
Looks good. But it seems that it isn't activated in the training yet.

Also, I don't think you need to return the optimizer. have a look at the [Imagenet example](https://github.com/pytorch/examples/blob/master/imagenet/main.py#L128)

I was thinking of keeping the training script clean and perhaps having the lr_scheduling and pytorch-specific utils in a separate module.

**Comment by ibadami** [comment]
I have updated the pull request by separating out the definition of the lr_scheduler into new module with the same name and activated it in the training.
I also added the imagenet scheduler into this module for future use.
